state/status.go

```diff
@@ -338,17 +384,81 @@ func PruneStatusHistory(st *State, maxHistoryTime time.Duration, maxHistoryMB in
 	if sizePerStatus == 0 {
 		return errors.New("unexpected result calculating status history entry size")
 	}
-	deleteStatuses := count - int(float64(collMB-maxHistoryMB)/sizePerStatus)
-	result := historicalStatusDoc{}
-	err = history.Find(nil).Sort("-updated").Skip(deleteStatuses).One(&result)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	_, err = history.RemoveAll(bson.D{
-		{"updated", bson.M{"$lt": result.Updated}},
+	toDelete := int(float64(collMB-p.maxSize) / sizePerStatus)
```

Comment:
I'm not sure whether we should bother working out how many we're expecting to delete. It's potentially useful for logging how much left there is to go (if there's a really large amount to prune), but it's misleading (hence the awkward "(estimated)" in the log message). We check the size after each batch, and stop once we're under the threshold (this seems important in case something else is also deleting history - like `unit.Destroy`), and generally this will be fewer than the calculation.